src/net.cpp

```diff
@@ -594,7 +594,7 @@ bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes)
         if (handled < 0)
                 return false;
 
-        if (msg.in_data && msg.hdr.nMessageSize > MAX_PROTOCOL_MESSAGE_LENGTH) {
+        if (msg.in_data && msg.hdr.nMessageSize > ((GetAdjustedTime() > (Params().ProtocolCleanupActivationTime() - 2*60*60 /* two hours */)) ? 0x7f000000 /* MAX_BLOCKFILE_SIZE */ : MAX_PROTOCOL_MESSAGE_LENGTH)) {
```

Comment:
I think this deserves being split into two lines. Also not sure why you aren't actually using the constant, but rather commenting about it.
```suggestion
        const bool use_blockfile_size = GetAdjustedTime() > (Params().ProtocolCleanupActivationTime() - 2*60*60 /* two hours */);
        if (msg.in_data && msg.hdr.nMessageSize > (use_blockfile_size ? 0x7f000000 /* MAX_BLOCKFILE_SIZE */ : MAX_PROTOCOL_MESSAGE_LENGTH)) {
```

Reply:
Because `main.h` isn't available to `net.cpp`. I could change that, but it'd be a layer violating to the architectural decisions already made by the bitcoin developers.

Reply:
Sounds like moving the definition to a more neutral place is warranted, in that case..

Reply:
I'm hesitant to do that for now, for the purpose of minimizing code difference from bitcoin. Maybe once I'm caught up on the rebases I will. Resolving this for now.

Reply:
Actually, I noticed that net.cpp includes primitives/block.h, so I moved the `MAX_BLOCKFILE_SIZE` constant there. Later it'll get moved into the consensus headers which are introduced in more recent versions of bitcoin. This really is a consensus rule since the client is incapable of processing a block larger than the maximum block file size.

In performing a final review of this PR, I'm also uncomfortable with the memory exhaustion DoS possibilities this presents after activation. I therefore redid this code to provide a smaller limit for 32-bit nodes based on the maximum number of network connections. Yes this makes a consensus rule based on a configurable parameter (shock!!), but only for 32-bit peers. Such nodes with default configurations will get pushed off the network if/when blocks grow larger than about 16MiB, but should a 32-bit node need to be operated past the protocol activation date (e.g. for testing purposes), it can be configured to use a higher limit by limiting the maximum number of inbound peers.

Obviously when forward blocks is deployed the actual limits will be used, even though by that time I suspect we will have dropped support for 32-bit peers entirely.